File: MapA7/MapA7/Model/Statements/CompoundStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model.Statements
{
    public class CompoundStatement:IStatement
    {
        private IStatement left;
        private IStatement right;

        public CompoundStatement(IStatement left, IStatement right)
        {
            this.left = left;
            this.right = right;
        }

        public ProgramState Execute(ProgramState ps)
        {
            ExecutionStack executionStack = ps.GetExecutionStack();
            executionStack.Push(right);
            executionStack.Push(left);
            return ps;
        }

        public String Tostring() { return left.Tostring() + ';' + right.Tostring(); }
    }
}

File: MapGUI3/src/Model/Statements/CompoundStatement.java
package Model.Statements;

import Model.ProgramState;
import Model.ExecutionStack;

public class CompoundStatement implements IStatement {
    private IStatement firstStatement;
    private IStatement secondStatement;

    public CompoundStatement(IStatement firstStatement, IStatement secondStatement) {
        this.firstStatement = firstStatement;
        this.secondStatement = secondStatement;
    }

    public ProgramState execute(ProgramState ps) {
        ExecutionStack<IStatement> executionStack = ps.getExecutionStack();
        executionStack.push(secondStatement);
        executionStack.push(firstStatement);
        return ps;
    }

    public String toString() {
        return firstStatement.toString() + "; " + secondStatement.toString();
    }
}
